react/features/base/storage/PersistenceRegistry.js

```diff
@@ -150,15 +168,23 @@ class PersistenceRegistry {
      *
      * @private
      * @param {Object} subtree - The redux state subtree.
-     * @param {Object} subtreeConfig - The related config.
+     * @param {ElementConfig} subtreeConfig - The related config.
      * @returns {Object}
      */
     _getFilteredSubtree(subtree, subtreeConfig) {
-        const filteredSubtree = {};
-
-        for (const persistedKey of Object.keys(subtree)) {
-            if (subtreeConfig[persistedKey]) {
-                filteredSubtree[persistedKey] = subtree[persistedKey];
+        let filteredSubtree;
+
+        if (typeof subtreeConfig === 'boolean' && subtreeConfig) {
+            // we persist the entire subtree
+            filteredSubtree = subtree;
+        } else if (typeof subtreeConfig === 'object') {
+            // only a filtered subtree gets persisted, based on the
+            // subtreeConfig object.
+            filteredSubtree = {};
+            for (const persistedKey of Object.keys(subtree)) {
+                if (subtreeConfig[persistedKey]) {
```

Comment:
wouldn't this skip keys which have a "falsy" value?

Reply:
Yes, it would, but it suppose to. Or do I misunderstand the question?

Reply:
but why? if a key we want to persist is some "enabled" flag, I still want to persist it, even if the current value is false. Would this case work? (maybe I'm confused, apologies in that case!)

Reply:
Ah, sorry, I misunderstood you. You're right, I'll change it

Reply:
No, sorry, I double-misunderstood you :) So this is just the subtree config key, which is, if false, we shouldn't persist it. It's not the subtree value, that gets checked. So if there is a boolean flag in the state, that gets persisted, regardless of the state of that, but if the subtreeConfig[key] is a false-like value, we shouldn't persist (and rehydrate) it.